### PR TITLE
feat(babel): rewrite for web (attempt 2)

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Alternative to https://github.com/expo/expo/pull/24725 which takes a less radical approach.
- Instead of reimplementing all the functionality, we'll simply remove the unused babel transforms from the existing preset.

# Test Plan

- Hopefully all language tests catch any changes.
- This only applies to web and ssr.

### CODE REVIEW

File: packages/babel-preset-expo/src/web-preset.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+const defaultPlugins = [
+  [require('@babel/plugin-syntax-flow')],
+  [require('babel-plugin-transform-flow-enums')],
+  [require('@babel/plugin-proposal-class-properties'), { loose }],
+  [require('@babel/plugin-transform-private-methods'), { loose }],
+  [require('@babel/plugin-transform-private-property-in-object'), { loose }],
+  [require('@babel/plugin-syntax-dynamic-import')],
+  [require('@babel/plugin-syntax-export-default-from')],
+  [require('@babel/plugin-syntax-nullish-coalescing-operator')],
+  [require('@babel/plugin-syntax-optional-chaining')],
```
Comment (ide) [inline packages/babel-preset-expo/src/web-preset.ts:24]: Are these still needed? Given that this PR removes the transform for `??=`, I'd imagine that `?.` and `??` don't need the syntax plugins either. Or `@babel/plugin-syntax-logical-assignment-operators` is needed here. But ideally Babel would parse modern ratified JS without the plugins.

File: packages/babel-preset-expo/src/web-preset.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+ * https://github.com/facebook/react-native/blob/2af1da42ff517232f1309efed7565fe9ddbbac77/packages/react-native-babel-preset/src/configs/main.js#L1
+ */
+
+// use `this.foo = bar` instead of `this.defineProperty('foo', ...)`
+const loose = true;
+
+const defaultPlugins = [
+  [require('@babel/plugin-syntax-flow')],
+  [require('babel-plugin-transform-flow-enums')],
+  [require('@babel/plugin-proposal-class-properties'), { loose }],
```
Comment (ide) [inline packages/babel-preset-expo/src/web-preset.ts:18]:
```suggestion
  [require('@babel/plugin-transform-class-properties'), { loose }],
```

However, is this needed? There are other removed transforms that imply supporting iOS 14 and up as the cutoff point, and class instance fields were added in iOS 14.5. Being clear on what the cutoff is will be important for anyone using this transform.

**Review by ide — CHANGES_REQUESTED** [review]

File: packages/babel-preset-expo/src/web-preset.ts (hunk 2 above)
Comment (EvanBacon) [reply packages/babel-preset-expo/src/web-preset.ts:18]: iOS 13.4 is the cutoff on native. We don't really have definitive rules for the web preset yet.

Comment (ide) [reply packages/babel-preset-expo/src/web-preset.ts:18]: OK - may need the syntax transform for logical assignment operators then

File: packages/babel-preset-expo/src/web-preset.ts (hunk 1 above)
Comment (EvanBacon) [reply packages/babel-preset-expo/src/web-preset.ts]: These don't appear to be needed anymore.

**Review by ide — APPROVED** [review]

**Pull request merged**